mm2src/coins/utxo.rs

```diff
@@ -1903,15 +1901,13 @@ where
     Ok(signed)
 }
 
-pub fn output_script(address: &Address, script_type: ScriptType) -> Script {
-    match address.addr_format {
-        UtxoAddressFormat::Segwit => Builder::build_p2witness(&address.hash),
-        _ => match script_type {
-            ScriptType::P2PKH => Builder::build_p2pkh(&address.hash),
-            ScriptType::P2SH => Builder::build_p2sh(&address.hash),
-            ScriptType::P2WPKH => Builder::build_p2witness(&address.hash),
-            ScriptType::P2WSH => Builder::build_p2witness(&address.hash),
-        },
+/// Builds transaction output script for an Address struct
+pub fn output_script(address: &Address) -> Script {
+    match address.script_type() {
+        AddressScriptType::P2PKH => Builder::build_p2pkh(address.hash()),
+        AddressScriptType::P2SH => Builder::build_p2sh(address.hash()),
+        AddressScriptType::P2WPKH => Builder::build_p2wpkh(address.hash()).expect("valid p2wpkh"),
+        AddressScriptType::P2WSH => Builder::build_p2wsh(address.hash()).expect("valid p2wsh"),
```

Comment:
A missed dev error here will lead to panicking which is not good but better than building the wrong output script and sending to a wrong address. I would say the best solution is always to propagate the errors.

Reply:
Added error propagation in [354c970](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1960/commits/354c970e3409d0e31629e40337b5d9819891adcd) - this led to not a few changes in the callers though